### PR TITLE
More material `docImport`s

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
https://github.com/flutter/flutter/issues/150800.

After applying the patch:
```
Can not fully resolve ../../flutter/packages/flutter/lib/src/material/material_state.dart: (MaterialState.pressed, MaterialState.focused, MaterialState.hovered, ..., value, update)
Can not fully resolve ../../flutter/packages/flutter/lib/src/material/material_state_mixin.dart: (MaterialStateProperty.resolve, MaterialState.disabled, MaterialState.dragged, ..., MaterialState.scrolledUnder, MaterialState.selected)
Can not fully resolve ../../flutter/packages/flutter/lib/src/material/reorderable_list.dart: (TargetPlatformVariant.desktop, TargetPlatformVariant.mobile)
Can not fully resolve ../../flutter/packages/flutter/lib/src/material/app.dart: (GlobalMaterialLocalizations)
Can not fully resolve ../../flutter/packages/flutter/lib/src/material/material_localizations.dart: (GlobalMaterialLocalizations)
57 out-of-scope references in 5 files, 22 unique symbols were left unresolved.
```

I'm going to leave `material_state` stuff alone because of https://github.com/flutter/flutter/issues/150800#issuecomment-2243923869, 
`TargetPlatformVariant.desktop` requires `flutter_test` and `GlobalMaterialLocalizations` requires `package:flutter_localizations/flutter_localizations.dart`.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/lib/src/material/about.dart [hunk 1]
```diff
@@ -1065,7 +1064,7 @@ class _MasterDetailFlow extends StatefulWidget {
 
   /// Builder for the master view for lateral navigation.
   ///
-  /// If [masterPageBuilder] is not supplied the master page required for nested navigation, also
+  /// This builder builds the master page required for nested navigation, also
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/material/about.dart:1067]: It appears the other parameter was never used, and it was removed in https://github.com/flutter/flutter/pull/96553

File: packages/flutter/lib/src/material/carousel.dart [hunk 2]
```diff
@@ -1261,7 +1261,8 @@ class _CarouselMetrics extends FixedScrollMetrics {
 
   /// The fraction of the viewport that the first item occupies.
   ///
-  /// Used to compute [item] from the current [pixels].
+  /// Used to compute the extent of each carousel item from the current [pixels],
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/material/carousel.dart:1264]: @QuncCccccc could you take a look and see if this makes sense?

File: packages/flutter/lib/src/material/date_picker.dart [hunk 3]
```diff
@@ -807,9 +807,6 @@ class _RestorableAutovalidateMode extends RestorableValue<AutovalidateMode> {
 /// * Single Date picker with calendar mode.
 /// * Single Date picker with text input mode.
 /// * Date Range picker with text input mode.
-///
-/// [helpText], [orientation], [icon], [onIconPressed] are required and must be
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/material/date_picker.dart:811]: Most of these are no longer relevant. And this should be obvious from the signature.

File: packages/flutter/lib/src/material/carousel.dart (hunk 2 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/carousel.dart:1264]: Yes, this makes sense to me. Thanks for also fixing the typos:)

**Comment by matanlurey** [comment]
test-exempt: Doc comments only.

File: packages/flutter/lib/src/material/about.dart [hunk 4]
```diff
@@ -1047,7 +1047,6 @@ enum _Focus { master, detail }
 
 /// A Master Detail Flow widget. Depending on screen width it builds either a
 /// lateral or nested navigation flow between a master view and a detail page.
-/// bloc pattern.
```
Comment (srawlins) [inline packages/flutter/lib/src/material/about.dart:1050]: Thanks for all the fixes!

**Review by srawlins — APPROVED** [review]
So many good fixes! This is gold!

**Pull request merged**